File: js/youtube.js
const fs = require("fs");

const API_KEY = fs.readFileSync("API_KEY", "utf8").trim();

async function search(type, part, query) {
  const url = `https://www.googleapis.com/youtube/v3/${type}`;
  const params = new URLSearchParams({
    part: part,
    key: API_KEY,
    q: query,
    chart: "mostPopular",
  });

  const response = await fetch(`${url}?${params}`);
  const data = await response.json();
  return data.items;
}

async function main() {
  try {
    const result = await search("videos", "statistics", "Python");
    // const result = await search("videos", "snippet", "Python");

    result.forEach((item) => {
      console.log("Link : " + "https://www.youtube.com/watch?v=" + item.id);
      // console.log("Title : " + item.snippet.title);
      console.log("View count: " + item.statistics.viewCount);
      console.log("---------------------------------");
    });
  } catch (error) {
    console.error("Error:", error);
  }
}

main();

File: py/youtube.py
import requests

API_KEY = open("API_KEY").read()
SEARCH_ENGINE_ID = open("SEARCH_ENGINE_ID").read()

def search(type,part,query):
    url = f"https://www.googleapis.com/youtube/v3/{type}"
    params = {
        "part": part,
        "key": API_KEY,
        "q": query,
        "chart": "mostPopular",
    }

    response = requests.get(url, params=params)
    data = response.json()["items"]
    return data

result = search("videos",["statistics", "snippet"],"Python")
# result = search("videos","snippet","Python")

for item in result:
    print("Link : " + "https://www.youtube.com/watch?v=" + item["id"])
    print("Title : " + item["snippet"]["title"])
    print("View count: " + item["statistics"]["viewCount"])
    print("---------------------------------")